includes/CLI.php

```diff
@@ -159,29 +159,38 @@ public static function coupons( $args, $assoc_args ) {
 
 		$time_start = microtime( true );
 
-		$min = 5;
-		$max = 100;
-		if ( ! empty( $assoc_args['min'] ) ) {
-			$min = $assoc_args['min'];
-		}
-		if ( ! empty( $assoc_args['max'] ) ) {
-			$max = $assoc_args['max'];
-		}
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating coupons', $amount );
 
-		if ( $amount > 0 ) {
-			$progress = \WP_CLI\Utils\make_progress_bar( 'Generating coupons', $amount );
-			for ( $i = 1; $i <= $amount; $i++ ) {
-				Generator\Coupon::generate( true, $min, $max );
+		add_action(
+			'smoothgenerator_coupon_generated',
+			function () use ( $progress ) {
 				$progress->tick();
 			}
-			$progress->finish();
+		);
+
+		$remaining_amount = $amount;
+		$generated        = 0;
+
+		while ( $remaining_amount > 0 ) {
+			$batch = $remaining_amount > Generator\Coupon::MAX_BATCH_SIZE ? Generator\Coupon::MAX_BATCH_SIZE : $remaining_amount;
+
+			$result = Generator\Coupon::batch( $batch, $assoc_args );
+
+			if ( is_wp_error( $result ) ) {
+				WP_CLI::error( $result );
+			}
+
+			$generated        += count( $result );
+			$remaining_amount -= $batch;
 		}
 
+		$progress->finish();
+
 		$time_end       = microtime( true );
 		$execution_time = round( ( $time_end - $time_start ), 2 );
 		$display_time   = $execution_time < 60 ? $execution_time . ' seconds' : human_time_diff( $time_start, $time_end );
 
-		WP_CLI::success( $amount . ' coupons generated in ' . $display_time );
+		WP_CLI::success( $generated . ' coupons generated in ' . $display_time );
```

Comment:
Passing observation, doesn't make sense to change this here (because it's an existing *thing* across the codebase, and it isn't what this change is focused on) but, at some point, it would be nice to internationalize.

Reply:
I've thought about that a few times. I think if we were distributing this plugin via wordpress.org, it might make sense to do that, because then we'd have the built-in infrastructure to get community translations, language packs, etc. But in the absence of any plans to do that, I'm not sure it's worth the effort to make the switch.

Also, the main way this plugin is used is via WP-CLI. Would we need to add translated aliases of the commands? (I actually don't know if wp-cli does this for any of the core commands.)

One other part is that we rely on some 3rd party composer packages for generating the random product names and such, and it's unclear how translatable those are. So even if we got the UI translated, it might still be generating products with English names (or Latin, in some cases).